.github/workflows/trigger-circle-ci-workflow.yml

```diff
@@ -2,20 +2,78 @@ name: Trigger CircleCI workflow
 
 on:
   pull_request:
-    types: [labeled]
+    types: [opened, synchronize, labeled, unlabeled, reopened, converted_to_draft, ready_for_review]
+  push:
 
 jobs:
-  trigger:
-    if: github.event.label.name == 'run e2e extended test suite' && github.event.pull_request.head.repo.fork == false
-    name: Run workflow with all e2e tests
+  trigger-ci-tests:
     runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request' && github.event.pull_request.draft == true && !contains(github.event.pull_request.labels.*.name, 'ci:pr') && !contains(github.event.pull_request.labels.*.name, 'ci:merged') && !contains(github.event.pull_request.labels.*.name, 'ci:daily')
```

Comment:
So people should only be able to use the labels once the PR is ready for review?